### PR TITLE
fix(deps): make sure pyarrow is not an implicit dependency

### PR DESCRIPTION
**Pull request by cpcloud**
This PR removes pyarrow as test dependency which was hiding an implicit postgres backend dependent.

### CODE REVIEW

File: ibis/backends/tests/test_register.py [hunk 1]
```diff
@@ -341,6 +360,7 @@ def test_csv_reregister_schema(con, tmp_path):
     ]
 )
 def test_register_garbage(con):
+    sa = pytest.importorskip("sa")
```
Comment (gforsyth) [inline ibis/backends/tests/test_register.py:363]:
```suggestion
    sa = pytest.importorskip("sqlalchemy")
```

**Review by gforsyth — CHANGES_REQUESTED** [review]
Looks good, just the one change, I think.

File: ibis/backends/tests/test_register.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/tests/test_register.py:363]: Thank you.

**Comment by gforsyth** [comment]
I think the mssql error is it being flaky on login, but the impala failure looks like an issue with the reconfigured test setup

File: ibis/backends/impala/tests/conftest.py [hunk 2]
```diff
@@ -152,12 +152,7 @@ def connect(
         )
 
     def _get_original_column_names(self, tablename: str) -> list[str]:
-        import pyarrow.parquet as pq
-
-        pq_file = pq.ParquetFile(
-            self.data_directory / "parquet" / tablename / f"{tablename}.parquet"
-        )
-        return pq_file.schema.names
+        return list(TEST_TABLES[tablename].keys())
```
Comment (gforsyth) [inline ibis/backends/impala/tests/conftest.py:155]:
```suggestion
        return list(TEST_TABLES[tablename].names)
```
?

**Pull request merged**